packages/aws-cdk-lib/aws-stepfunctions/README.md

```diff
@@ -562,6 +562,10 @@ Custom states can be chained together with any of the other states to create you
 definition. You will also need to provide any permissions that are required to the `role` that
 the State Machine uses.
 
+The Retry and Catch fields are available for error handling.
+You can configure the Retry field by defining it in the JSON object or by adding it using the `addRetry` method.
+However, the Catch field cannot be configured by defining it in the JSON object, so it must be added using the `addCatch` method.
```

Comment:
why not to do the same for the Catch field?

Reply:
@moelasmar 
Thanks for clarifying.
As mentioned in the comments (https://github.com/aws/aws-cdk/issues/25798#issuecomment-1749916284), the method of defining the `Catch` directly in JSON for the `CustomState` did not render well and forced us to call `_addCatch` in the base `State` class.
The `addCatch` method was added to solve this (https://github.com/aws/aws-cdk/pull/28422), but now it overrides the `Catch` and `Retry` fields in the JSON.
So it will take some investigation and changes to be able to define the `Catch` directly in JSON.

Reply:
thanks @sakurai-ryo for the clarifying.